stripe/src/main/java/com/stripe/android/paymentsheet/flowcontroller/FlowControllerInitializer.kt

```diff
@@ -0,0 +1,27 @@
+package com.stripe.android.paymentsheet.flowcontroller
+
+import com.stripe.android.paymentsheet.PaymentSheet
+
+/**
+ * An interface for a class that can initialize a [PaymentSheet.FlowController].
+ */
+internal interface FlowControllerInitializer {
+    suspend fun init(
+        paymentIntentClientSecret: String,
+        configuration: PaymentSheet.Configuration
+    ): InitResult
+
+    suspend fun init(
+        paymentIntentClientSecret: String
+    ): InitResult
+
+    sealed class InitResult {
+        data class Success(
+            val initData: InitData
+        ) : InitResult()
+
+        class Failure(
+            val throwable: Throwable
```

Comment:
Is it possible to fail without having an error attached (ie, should this be nullable)?

Reply:
We'll always have an error